viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -0,0 +1,286 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewer, Intersection } from '@reveal/core';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import * as THREE from 'three';
+import { MeasurementLabel } from './MeasurementLabel';
+import { MeasurementLineOptions, MeasurementOptions, MeasurementUnits, MeasurementUnitUpdateDelegate } from './types';
+import { LineGeometry } from 'three/examples/jsm/lines/LineGeometry';
+import { LineMaterial } from 'three/examples/jsm/lines/LineMaterial';
+
+export class MeasurementTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _measurementLabel: MeasurementLabel;
+  private _lineGeometry: LineGeometry;
+  private _lineMesh: THREE.Mesh;
+  private _lineMaterial: LineMaterial;
+  private readonly _linePosition: Float32Array;
+  private readonly _raycaster: THREE.Raycaster;
```

Comment:
Seems a little overkill to use a raycaster here since you are never actually ray casting anything, just using it to construct a line, please correct me if I misunderstood anything.